app/views/events/participants.html.haml

```diff
@@ -0,0 +1,24 @@
+.col-md-12
+  %h3 Participants of #{@event.name}
+  %br
+  %table.table
+    %tr
+      %th #
+      %th Nickname
+      %th Email
+      %th State
+      %th Role
+      %th Change role to event organizer
+    - @event.users.includes(:profile).distinct.each.with_index(1) do |user,index|
```

Comment:
But since within the loop we still query the requests, you might want to include the requests here.